agent/src/agent/pipeline/manager.py

```diff
@@ -169,35 +156,52 @@ def edit_using_json(configs: list) -> List[Pipeline]:
     pipelines = []
     for config in configs:
         try:
-            pipelines.append(
-                edit_pipeline_using_json(config)
-            )
+            pipelines.append(edit_pipeline_using_json(config))
         except Exception as e:
             exceptions[config['pipeline_id']] = f'{type(e).__name__}: {str(e)}'
         if exceptions:
             raise pipeline.PipelineException(json.dumps(exceptions))
     return pipelines
 
 
+class PipelineWatcher:
```

Comment:
yes. There are two places now. Ideally, we should have one place where that check will happen.
To have a method Pipeline.config_changed() is not good either, it adds one more responsibility to Pipeline and you can call that method anytime but it will work only when you called set_config first
When I changed code a bit I got an error when calling config_changed() because when you get a pipeline from the db it doesn't have _previous_config, nor _orevious_override_source. That's why I removed it

Reply:
why not just fix the config_changed method then? If set_config wasn't called and we just queried the object from the db then we should just return False, right?